### DISCUSSION TITLE
PlutusData

### DISCUSSION THREAD

**Discussion by martap79** [Python-Cardano/pycardano#55] [Q&A]
Hi!, could we pls get some example code how to use PlutusData? various ways of reading in the datum, either cbor, dict or json;
I've been trying to spend a script utxo but can't get the datum in. I've tried using the from_dict, from_primitive and from_json but plutusdata object stays empty. I'm obviously doing something wrong, but can't figure it out.

**Comment by cffls** [comment]
Hi @martap79 , have you read [this guide](https://pycardano.readthedocs.io/en/latest/guides/plutus.html#datum-and-redeemer-serialization) about Plutus? It shows some examples of creating datum. Also take a look at plutus test file [here](https://github.com/cffls/pycardano/blob/main/test/pycardano/test_plutus.py), which contains more examples of constructing datums.
